posts/2019/12/Spyder-terminal.md

```diff
@@ -0,0 +1,54 @@
+<!--
+.. title: Creating the ultimate terminal experience in Spyder 4 with spyder-terminal
+.. slug: creating-the-ultimate-terminal-experience-in-Spyder-4-with-spyder-terminal
+.. date: 2019-12-17 16:00:00 UTC-05:00
+.. author: Stephannie Jimenez
+.. tags: Labs, Spyder
+.. category:
+.. link:
+.. description:
+.. type: text
+-->
+
+The Spyder-terminal project is revitalized! In the new release of the project we were able to add numerous features that improve the terminal experience. As well, this new version enhances compatibility with the latest Spyder 4 release. We were able to develop all this work thanks to the latest release of the [xterm.js](https://xtermjs.org/) project.
+
+<!-- TEASER_END -->
+
+## ES6/JSX syntax
+In the first place, we were able to update all the old javascript files to a ES6/JS syntax. This change simplified the code base and maintenance for the terminal. Also, it allows us to extend the project in an easier way to new functionalities that the xtermjs api offer. In order to compile this code and run it inside Spyder we migrated our deployment to Webpack. Additionally, the tests for the client terminal were updated to support the new syntax.
+
+## Multiple shells per operative system
+
+A new feature added to the terminal is that the user can configure which shell to use in the terminal. Now the terminal is able to start with 'bash', 'sh', 'ksh', 'zsh', 'csh', 'pwsh', 'tcsh', 'screen', 'tmux', 'dash' and 'rbash' in UNIX systems and 'cmd' and 'powershell' in Windows. This option is given on the preferences pane and a restart of Spyder is required to apply the changes.
+
+![UNIX shell options for starting the terminal](/images/spyder-terminal/shells.png)
+
+This is a great feature because it allows the user to determine their shell interpreter among the ones that are installed in their systems. In this way, Spyder Terminal can be configured with any of the existing shells as long as it is available on their machine. Some examples of the configurable options of the interpreter are shown below, with bash and tcsh.
+
```

Comment:
These two screenshots show empty terminals, I don't see the configurable options. Wrong images?

Reply:
The previous screenshot shows the options that are available in the preferences pane. While these two screenshots display the terminal under two of the configurable shells, I'm not sure if they are necessary.

Reply:
Yeah, probably better to remove the screenshots and the sentence that mentions them, I don't think it adds too much.